packages/next/src/client/script.tsx

```diff
@@ -127,6 +147,11 @@ const loadScript = (props: ScriptProps): void => {
 
   el.setAttribute('data-nscript', strategy)
 
+  // Load styles associated with this script
+  if (stylesheets) {
+    insertStylesheets(stylesheets)
+  }
```

Comment:
Forgive me if I'm misunderstanding, but wouldn't this run every time since you're not checking for `!appDir` here? Could it make more sense to move this method call to later in the file where you are checking for `appDir` (line 296):

```js
if (appDir) {
  let stylesToRender = null
  let scriptsToRender = null
  // ...
} else {
    if (stylesheets) insertStylesheets(stylesheets)
}
```

If this makes sense, better yet to move the stylesheets conditional to be first. So:

```js
if (stylesheets) {
  if (appDir) {
    let stylesToRender = null
    let scriptsToRender = null
    // ...
  } else {
    insertStylesheets(stylesheets)
  }
}
```

Let me know if there's a reason this can't be done though.

Reply:
I check if appDir and mark this undefined [here](https://github.com/vercel/next.js/pull/51194/files#diff-d7794cd766752e9cdff323db1f142a20fc6ac348eb399d0ec5f600ed122d4b0eR216)
So this condition will never execute if appDir is enabled.

We can't move the logic to insertStylesheet outside the loadScript function, because the script can be loaded from different timings depending on strategy. And we want to load styles only if the script loads. the loadScript function checks cache and only loads the script once.

Reply:
Ah got it, that makes total sense :)

Thanks for clarifying 🙏